modules/standard/Types.chpl

```diff
@@ -711,4 +711,110 @@ inline proc _bor_id(type t) return 0:t;
 pragma "no doc"
 inline proc _bxor_id(type t) return 0:t;
 
+
+/* Return the number of fields in a class or record as a param.
+   The count of fields includes types and param fields.
+ */
+proc numFields(type t) param
```

Comment:
I realize that you're arguably following the already-established mode used by this file, but I think that in the spirit of making the documentation as clear as possible (and supporting good interfaces), we should declare the return types of library routines whenever possible.  I.e., I'd change the return type to 'param : int' (and, if I'd reviewed the original type documentation, would've required 'param : bool' for all the is*Type queries -- I'm not asking you to change the existing code, though).  I think the getField* routines are the main ones that can't do this because the module code can't assert what the return type is?